book/src/lists.md

```diff
@@ -0,0 +1,210 @@
+# Lists
+
+Sometimes you want to display a list of elements in a certain arrangement.
+[`ListBox`](../docs/gtk4/struct.ListBox.html) and [`FlowBox`](../docs/gtk4/struct.FlowBox.html) are two container widgets which allow to do this.
+`ListBox` describes a vertical list and `FlowBox` describes a grid.
+
+Let us explore this concept by adding labels to a `ListBox`.
+Each label will display an integer starting from 0 and ranging up to 100.  
+
+<span class="filename">Filename: listings/lists/1/main.rs</span>
+
+```rust,no_run
+{{#rustdoc_include ../listings/lists/1/main.rs:list_box}}
+```
+
+We cannot display so many widgets at once.
+Therefore, we add `ListBox` to a [`ScrolledWindow`](../docs/gtk/struct.ScrolledWindow.html).
+Now we can scroll through our elements.
+
+<span class="filename">Filename: listings/lists/1/main.rs</span>
+
+```rust,no_run
+{{#rustdoc_include ../listings/lists/1/main.rs:scrolled_window}}
+```
+
+<div style="text-align:center"><img src="img/lists_list_box.png"/></div>
+
+That was easy enough.
+However, we currently create one widget per element.
+Since creating new widgets takes a bit of time, this can become a problem with many elements.
+`ListBox` can easily handle a few thousands of them, but how could we possibly deal with the infinite amount of posts in a social media timeline?
```

Comment:
Technically it's not ListBox that can handle thousands of elements, it's the creation of thousands of widgets that can be fine if those widgets are simple enough such as labels.  otherwise seems fine :)